### ISSUE TITLE
[BUG] Unable to upsert kind 0 event

### ISSUE THREAD

**Issue by Cameri** [Cameri/nostream#122]
**Describe the bug**
Some replaceable events are unable to be saved to the database. Likely having to do with the deduplication field.

**To Reproduce**
1. Publish the following event:
```json
   {
     "pubkey": "75cf971c22dc7694192ca45ba9951636743e35ae5d45dbfc03ceb24cb51df57d",
     "created_at": 1672587032,
     "kind": 0,
     "tags": [],
     "content": "{\"name\":\"ai57783266\",\"picture\":\"https://pbs.twimg.com/profile_images/1518535461433528321/pAGPg2LH_400x400.jpg\"}",
     "id": "9bdef93b60c24f96b38b4937937e9a4f43efdf316ee9e95dfdf91c94e4c11a7b",
     "sig": "3d7476610b988983365ccb14f3e672e274c2bcfb338cc5a6c53c43ca0f0acb79f9cf4eb6c5a38eac733cd4ad6b4078ecf53c1b6d7e91312a8470a6fb451c5c4c"
   }
```

Valid as per: https://nostr-army-knife.netlify.app/
2. See error on the console.

**Expected behavior**
1. Event is either upserted or ignored.
2. No error is logged on the console.

**Screenshots**
![image](https://user-images.githubusercontent.com/378886/210182823-b7fce92c-021f-4257-a867-bb4398c86267.png)

**Logs**
If applicable, attach log messages showing the bug and context when the bug happened.
```
2023-01-01T15:30:33.995632813Z error handling message [
2023-01-01T15:30:33.995708060Z   'EVENT',
2023-01-01T15:30:33.995714311Z   {
2023-01-01T15:30:33.995718559Z     pubkey: '75cf971c22dc7694192ca45ba9951636743e35ae5d45dbfc03ceb24cb51df57d',
2023-01-01T15:30:33.995722957Z     created_at: 1672587032,
2023-01-01T15:30:33.995727580Z     kind: 0,
2023-01-01T15:30:33.995732033Z     tags: [],
2023-01-01T15:30:33.995736257Z     content: '{"name":"ai57783266","picture":"https://pbs.twimg.com/profile_images/1518535461433528321/pAGPg2LH_400x400.jpg"}',
2023-01-01T15:30:33.995741068Z     id: '9bdef93b60c24f96b38b4937937e9a4f43efdf316ee9e95dfdf91c94e4c11a7b',
2023-01-01T15:30:33.995745590Z     sig: '3d7476610b988983365ccb14f3e672e274c2bcfb338cc5a6c53c43ca0f0acb79f9cf4eb6c5a38eac733cd4ad6b4078ecf53c1b6d7e91312a8470a6fb451c5c4c'
2023-01-01T15:30:33.995750533Z   }
2023-01-01T15:30:33.995755879Z ] error: insert into "events" ("event_content", "event_created_at", "event_deduplication", "event_delegator", "event_id", "event_kind", "event_pubkey", "event_signature", "event_tags") values ($1, $2, $3, $4, $5, $6, $7, $8, $9) on conflict (event_pubkey, event_kind, event_deduplication) WHERE (event_kind = 0 OR event_kind = 3 OR event_kind = 41 OR (event_kind >= 10000 AND event_kind < 20000)) OR (event_kind >= 30000 AND event_kind < 40000) do update set "event_id" = $10,"event_created_at" = $11,"event_tags" = $12,"event_content" = $13,"event_signature" = $14,"event_delegator" = $15 where "events"."event_created_at" < $16 - duplicate key value violates unique constraint "events_event_id_unique"
2023-01-01T15:30:33.995764942Z     at Parser.parseErrorMessage (/app/node_modules/pg-protocol/dist/parser.js:287:98)
2023-01-01T15:30:33.995769161Z     at Parser.handlePacket (/app/node_modules/pg-protocol/dist/parser.js:126:29)
2023-01-01T15:30:33.995772819Z     at Parser.parse (/app/node_modules/pg-protocol/dist/parser.js:39:38)
2023-01-01T15:30:33.995776635Z     at Socket.<anonymous> (/app/node_modules/pg-protocol/dist/index.js:11:42)
2023-01-01T15:30:33.995780675Z     at Socket.emit (node:events:513:28)
2023-01-01T15:30:33.995784486Z     at addChunk (node:internal/streams/readable:324:12)
2023-01-01T15:30:33.995788253Z     at readableAddChunk (node:internal/streams/readable:297:9)
2023-01-01T15:30:33.995792275Z     at Readable.push (node:internal/streams/readable:234:10)
2023-01-01T15:30:33.995796529Z     at TCP.onStreamRead (node:internal/stream_base_commons:190:23) {
2023-01-01T15:30:33.995800924Z   length: 276,
2023-01-01T15:30:33.995805591Z   severity: 'ERROR',
2023-01-01T15:30:33.995809804Z   code: '23505',
2023-01-01T15:30:33.995813459Z   detail: 'Key (event_id)=(\\x9bdef93b60c24f96b38b4937937e9a4f43efdf316ee9e95dfdf91c94e4c11a7b) already exists.',
2023-01-01T15:30:33.995847074Z   hint: undefined,
2023-01-01T15:30:33.995851110Z   position: undefined,
2023-01-01T15:30:33.995856410Z   internalPosition: undefined,
2023-01-01T15:30:33.995860325Z   internalQuery: undefined,
2023-01-01T15:30:33.995863980Z   where: undefined,
2023-01-01T15:30:33.995867610Z   schema: 'public',
2023-01-01T15:30:33.995870957Z   table: 'events',
2023-01-01T15:30:33.995874338Z   column: undefined,
2023-01-01T15:30:33.995877949Z   dataType: undefined,
2023-01-01T15:30:33.995881229Z   constraint: 'events_event_id_unique',
2023-01-01T15:30:33.995884486Z   file: 'nbtinsert.c',
2023-01-01T15:30:33.995887881Z   line: '663',
2023-01-01T15:30:33.995891040Z   routine: '_bt_check_unique'
2023-01-01T15:30:33.995894563Z }
```

**Additional context**
DB Dump for this pubkey:
```
INSERT INTO events (id,event_id,event_pubkey,event_kind,event_created_at,event_content,event_tags,event_signature,first_seen,event_delegator,deleted_at,event_deduplication) VALUES
	 ('198fa24d-9bfa-4a62-8a63-b84b30642807',decode('9BDEF93B60C24F96B38B4937937E9A4F43EFDF316EE9E95DFDF91C94E4C11A7B','hex'),decode('75CF971C22DC7694192CA45BA9951636743E35AE5D45DBFC03CEB24CB51DF57D','hex'),0,1672587032,'{"name":"ai57783266","picture":"https://pbs.twimg.com/profile_images/1518535461433528321/pAGPg2LH_400x400.jpg"}','[]',decode('3D7476610B988983365CCB14F3E672E274C2BCFB338CC5A6C53C43CA0F0ACB79F9CF4EB6C5A38EAC733CD4AD6B4078ECF53C1B6D7E91312A8470A6FB451C5C4C','hex'),'2023-01-01 15:30:33.922042',NULL,NULL,'["75cf971c22dc7694192ca45ba9951636743e35ae5d45dbfc03ceb24cb51df57d", 0]'),
	 ('2dbec8bf-46c8-4d4c-a358-1d1bfe3cfbda',decode('891197D5BC6D30D81B8AEF6199C66B34DC82BD590BA0CEE5C6005608C5FE2E9E','hex'),decode('75CF971C22DC7694192CA45BA9951636743E35AE5D45DBFC03CEB24CB51DF57D','hex'),3,1672587032,'{"wss://nostr-pub.wellorder.net":{"read":true,"write":true},"wss://nostr.onsats.org":{"read":true,"write":true},"wss://nostr-relay.wlvs.space":{"read":true,"write":true},"wss://nostr.bitcoiner.social":{"read":true,"write":true},"wss://relay.damus.io":{"read":true,"write":true},"wss://nostr.zebedee.cloud":{"read":true,"write":false},"wss://relay.nostr.info":{"read":true,"write":false},"wss://nostr-pub.semisol.dev":{"read":true,"write":false},"wss://nostr.walletofsatoshi.com":{"read":true,"write":false}}','[]',decode('F7FA486E9226C9785974C6726A7D6E910014AA5C3DF2BFC38AC5D1EEBF48ED2E6451509BFF19680AB7B7A5A73A0FFFDABB43BF80D8E4817C64AEEA839C4D268D','hex'),'2023-01-01 15:30:33.997026',NULL,NULL,'["75cf971c22dc7694192ca45ba9951636743e35ae5d45dbfc03ceb24cb51df57d", 3]');
```

**Comment by etemiz** [comment]
is this relevant? 
https://www.postgresqltutorial.com/postgresql-tutorial/postgresql-upsert/

**Comment by Cameri** [comment]
@etemiz are you considering taking this bug?

**Comment by etemiz** [comment]
If this is time sensitive, I don't have a development environment for this.

**Comment by etemiz** [comment]
I set up a new machine for this. I posted the same event a few times. It did not give me the error. 
I am starting it like
`DEBUG=* npm run dev`

Maybe my row id's are different in the clean database?

Looking at the query, it has already gone pretty big. I would do some functionality at the application layer. This SQL query is not looking efficient and will be problem in the long run I think..

**Comment by Cameri** [comment]
solved in #134
